harper-core/src/spell/mod.rs

```diff
@@ -2,132 +2,118 @@ use std::borrow::Cow;
-
-    let pruned_words = words_to_search.filter_map(|word| {
-        let dist = edit_distance_min_alloc(&misspelled_word, word, &mut buf_a, &mut buf_b);
-        let dist_lower = edit_distance_min_alloc(&misspelled_lower, word, &mut buf_a, &mut buf_b);
-
-        if dist.min(dist_lower) <= max_edit_dist {
-            Some((word, dist))
-        } else {
-            None
-        }
-    });
-
-    // Locate the words with the lowest edit distance.
-    let mut found_dist: Vec<(&[char], u8)> = Vec::with_capacity(result_limit);
+#[derive(PartialEq)]
+pub struct FuzzyMatchResult<'a> {
+    word: &'a [char],
+    edit_distance: u8,
+    metadata: WordMetadata,
+}
 
-    for (word, dist) in pruned_words {
-        if found_dist.len() < result_limit {
-            found_dist.push((word, dist));
-        } else if dist < found_dist[result_limit - 1].1 {
-            found_dist[result_limit - 1] = (word, dist);
-        }
-        found_dist.sort_by_key(|a| a.1);
+impl<'a> PartialOrd for FuzzyMatchResult<'a> {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.edit_distance.partial_cmp(&other.edit_distance)
     }
+}
 
-    // Create final, ordered list of suggestions.
-    let mut found = Vec::with_capacity(found_dist.len());
-
+fn order_suggestions(matches: Vec<FuzzyMatchResult>) -> Vec<&[char]> {
+    let mut found: Vec<&FuzzyMatchResult> = Vec::with_capacity(matches.len());
     // Often the longest and the shortest words are the most helpful, so lets push
     // them first.
-    let minmax = found_dist.iter().position_minmax_by_key(|a| a.0.len());
+    let minmax = matches.iter().position_minmax_by_key(|fmr| fmr.word.len());
     if let MinMaxResult::MinMax(a, b) = minmax {
         if a == b {
-            found.push(found_dist[a].0);
+            found.push(&matches[a]);
         } else {
-            found.push(found_dist[a].0);
-            found.push(found_dist[b].0);
+            found.push(&matches[a]);
+            found.push(&matches[b]);
         }
 
         // Push the rest
         found.extend(
-            found_dist
-                .into_iter()
+            matches
+                .iter()
                 .enumerate()
                 .filter(|(i, _)| *i != a && *i != b)
-                .map(|v| v.1 .0),
+                .map(|v| v.1),
         );
     } else {
-        // Push the rest
-        found.extend(found_dist.into_iter().map(|v| v.0));
+        found.extend(&matches);
     }
 
     // Swap the lowest edit distance word with the shortest.
     if found.len() >= 3 {
         found.swap(0, 2);
     }
 
-    // Let common words bubble up.
-    found.sort_by_key(|v| {
-        if dictionary.get_word_metadata(v).common {
-            0
-        } else {
-            1
-        }
-    });
+    // Let common words bubble up, but do not prioritize them over all else.
+    found.sort_by_key(|fmr| fmr.edit_distance + if fmr.metadata.common { 0 } else { 1 });
```

Comment:
Why did you do this?

Reply:
I found we got better results this way. Before, any common word is prioritized from the hundred results, which doesn't make much sense.

If there are three common words out of the hundred results, then all of them would be prioritized over an uncommon word of edit distance 1.

This way, being uncommon just acts as if it were one further away.

I'm fine with getting rid of this if you want, iirc it was something I was trying when I was working on the `I'm` test, but it had good results so I kept it.

Reply:
Leave it as-is. It's on my radar and it's honestly low-priority.